tasks/apm-host-injection.yml

```diff
@@ -0,0 +1,35 @@
+---
+- name: Check if platform is supported
+  fail:
+    msg: APM Injection is not supported in this platform
+  when: ansible_facts.os_family not in ["Debian", "RedHat", "Rocky", "AlmaLinux"]
+
+- name: Flush handlers
+  meta: flush_handlers
+
+- name: Ensure datadog-agent is running
```

Comment:
Why does the Agent need to be running at this point?

If indeed it does need to be running, it should have `when: not ansible_check_mode`. You can learn more about the check mode here: https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_checkmode.html#using-check-mode

If it doesn't need to be running at this point, we can probably also remove the `flush_handlers` task.

Reply:
Official [doc](https://docs.datadoghq.com/tracing/trace_collection/library_injection_local/?tab=agentonhostappincontainers#install-the-preload-library) says it's necessary, but will confirm if it actually is 👍

Reply:
Confirmed it's not necessary for the agent to be running